src/main/java/seedu/condonery/logic/parser/ParserUtil.java

```diff
@@ -89,4 +90,34 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
         }
         return tagSet;
     }
+
+    /**
+     * Parses a {@code String clientName} into a {@code Client}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code clientName} does not match to a unique
+     * existing client's name.
+     */
+    public static Client parseClientName(String clientName) throws ParseException {
+        requireNonNull(clientName);
+        String trimmedClientName = clientName.trim();
+        return new Client(
```

Comment:
I'm not very sure if this function should return a 'Client'. I understand that this is being stored in the Property as interestedClients, but it's abit strange to create clients with substring names and placeholder addresses, I think I prefer if it stores the actual client instead.

I see that in the AddPropertyCommand you use the parser to find the actual references to the clients. Is it possible to just store these references to the clients directly in the Property?

Reply:
Thanks hr! I agree that it's kinda weird. There were some issues I faced and I thought I would just implement this in the simplest way possible first, then improve it in future iterations.
I haven't figured out how to solve those issues, but perhaps I could ask you for suggestions if you don't mind?

First, some considerations:
- The `ModelManager` object is required (e.g. AddPropertyCommand takes in a `Model` object in its `execute()` function) for finding matches to `Clients`, based on a user's input, since a `ModelManager` is initialised at the opening of Condonery, and a `ModelManager` holds reference to the most updated `ClientDirectory`
- Among all the involved classes in a typical command call, only the `AddPropertyCommand` class interacts with a `ModelManager` object. Thus, the matching of `Clients` to user input has to be done in the `AddPropertyCommand ` class


Basically, I have considered storing those references to the clients directly in the `Property`, but thought it would be a bit weird because with this then a `Property` would contain a
```java
Set<Client> interestedClients
```
as well as a
```java
Set<String> interestedClientNames // names found in the user's argument
```

I thought it would be weird because then Properties would directly hold reference to a user's input, while

Reply:
Thanks hr! I agree that it's kinda weird. There were some issues I faced and I thought I would just implement this in the simplest way possible first, then improve it in future iterations.
I haven't figured out how to solve those issues, but perhaps I could ask you for suggestions if you don't mind?

First, some considerations:
- The `ModelManager` object is required (e.g. AddPropertyCommand takes in a `Model` object in its `execute()` function) for finding matches to `Clients`, based on a user's input, since a `ModelManager` is initialised at the opening of Condonery, and a `ModelManager` holds reference to the most updated `ClientDirectory`
- Among all the involved classes in a typical command call, only the `AddPropertyCommand` class interacts with a `ModelManager` object. Thus, the matching of `Clients` to user input has to be done in the `AddPropertyCommand ` class
- Currently, I can only access the `ClientDirectory` in the `AddPropertyCommand` class. I decided to create a `Property` with the placeholder `interestedClients` first, before passing it into `AddPropertyCommand`. Thus, in `AddPropertyCommand`, I can access the user's input via the placeholder `interestedClients` in the `Property`, and also access the `ClientDirectory` through the `Model` object.

The issue is that I could not think of a better way to design this, as certain classes don't normally have access to certain data. 

One way to change this I think, is to make a new method in a `Property`: `addInterestedClients`, while the `AddPropertyCommand` would take in a new Property as well as the user's input. Then, the `AddPropertyCommand` would match `Clients` to the user input as usual before passing those `Clients` into the `addInterestedClients` of the new `Property`.

I hope I have been clear, let me know if there are parts that require more clarification. Overall, WDYT?

Reply:
ah i see what you mean! I was thinking instead of storing the `interestedClients` placeholder within the Property first, maybe we can store the `Set<String>` into the `AddPropertyCommand` class first. The reason being as currently it implies that the `Client` objects in Property are the real clients, and I'm afraid in the future we forget to update the references it stores. But by keeping the queries as strings, we avoid this confusion. 

Then when the `AddPropertyCommand` executes, we can do the same logic of querying the `Model`. I agree with the suggestion of `addInterestedClients`, and we can call this method before adding the `Property` to the directory. 

But yup! I think code quality wise it is okay to improve in the future! good to make sure the feature works first before we try to refactor it.